### PR TITLE
PSP-985 SonarQube cleanup

### PR DESCRIPTION
**Pull request by Fosol**
Updates to backend based on recommendations from SonarQube.

### CODE REVIEW

File: .github/workflows/api-dotnetcore.yml [hunk 1]
```diff
@@ -71,7 +71,7 @@ jobs:
           cd ..
           head TestResults/coverage.opencover.xml
       - name: Codecov
-        uses: codecov/codecov-action@v1.0.13
+        uses: codecov/codecov-action@v1.5.0
```
Comment (Fosol) [inline .github/workflows/api-dotnetcore.yml:74]: Fix GitHub Action failures due to older version no longer working.

File: .github/workflows/app-react.yml [hunk 2]
```diff
@@ -33,7 +33,7 @@ jobs:
       - run: npm run coverage
         working-directory: ${{env.working-directory}}
       - name: Codecov
-        uses: codecov/codecov-action@v1.0.13
+        uses: codecov/codecov-action@v1.5.0
```
Comment (Fosol) [inline .github/workflows/app-react.yml:36]: Fix GitHub Action failures due to older version no longer working.

File: backend/api/Configuration/ProgramOptions.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
+using System.Linq;
+using CommandLine;
+using Pims.Core.Extensions;
+
+namespace Pims.Api.Configuration
+{
+    /// <summary>
+    /// ProgramOptions class, provides a way to parse command line arguments for the PIMS API program.
+    /// </summary>
+    public class ProgramOptions
```
Comment (Fosol) [inline backend/api/Configuration/ProgramOptions.cs:10]: To ensure command line arguments are safe this provides us a way to parse them.

File: backend/api/Pims.Api.csproj [hunk 4]
```diff
@@ -18,6 +18,7 @@
     <PackageReference Include="AspNetCore.HealthChecks.UI.Client" Version="5.0.1" />
     <PackageReference Include="AspNetCore.HealthChecks.SqlServer" Version="5.0.3" />
     <PackageReference Include="AspNetCore.HealthChecks.UI" Version="5.0.1" />
+    <PackageReference Include="CommandLineParser" Version="2.8.0" />
```
Comment (Fosol) [inline backend/api/Pims.Api.csproj:21]: To ensure command line arguments are safe this provides us a way to parse them.

File: backend/api/Program.cs [hunk 5]
```diff
@@ -19,18 +22,31 @@ public class Program
         /// <param name="args"></param>
         public static void Main(string[] args)
         {
-            // ConfigureLogging();
-            var builder = CreateWebHostBuilder(args);
-            builder.Build().Run();
+            var results = Parser.Default.ParseArguments<ProgramOptions>(args);
```
Comment (Fosol) [inline backend/api/Program.cs:25]: To ensure command line arguments are safe this provides us a way to parse them.

**Review by devinleighsmith — APPROVED** [review]

**Review by jdtoombs — APPROVED** [review]
Doesn't look like we are stepping on each other's toes!

**Pull request merged**